### PR TITLE
Add oauth-provider-go to custom-authentication.md

### LINKED ISSUE #784: netlify-cms oauth in golang
Hi!
Inspired by [netlify-cms-github-oauth-provider](https://github.com/vencax/netlify-cms-github-oauth-provider) (node-js), and I created my implementation on Go lang.

See [netlify-cms-oauth-provider-go](https://github.com/igk1972/netlify-cms-oauth-provider-go)

Maybe it's interesting to someone.

### PR DESCRIPTION
**Pull request by igk1972**
Add netlify-cms-oauth-provider-go to docs/custom-authentication.md

EDIT: Closes #784.

### CODE REVIEW

File: docs/custom-authentication.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ Netlify CMS is meant to be platform agnostic, so we're always looking to expand
 | Author     | Supported Git hosts       | Languages | Link                                                                |
 |------------|---------------------------|-----------|---------------------------------------------------------------------|
 | @vencax    | GitHub, GitHub Enterprise | Node.js   | [Repo](https://github.com/vencax/netlify-cms-github-oauth-provider) |
+| @igk1972   | GitHub, GitHub Enterprise, Bitbacket, Gitlab | Go | [Repo](https://github.com/igk1972/netlify-cms-oauth-provider-go) |
```
Comment (tech4him1) [inline docs/custom-authentication.md]: I wonder if we should add a note below this table saying that BitBucket and GitLab are not yet supported in the CMS? Or just let people figure that out on their own? @verythorough?

**Comment by tech4him1** [comment]
Closes #784.

File: docs/custom-authentication.md (hunk 1 above)
Comment (igk1972) [reply docs/custom-authentication.md]: Little step to the future. Main keyword "...**not yet**...".

Edit line in docs/custom-authentication.md ?

Comment (erquhart) [reply docs/custom-authentication.md]: If it doesn't actually support those hosts, I'd say remove them for now.

**Comment by erquhart** [comment]
@igk1972 I usually don't fix merge conflicts on PR's submitted on the fork's master branch, but it seemed harmless to update yours - and then I pushed our master to your fork instead of the updated version of your master 🙄. Re-opened with your changes under #845.

**Comment by igk1972** [comment]
Ok, thanks

**Pull request closed without merging**